.github/workflows/msys2_ffmpeg.yml

```diff
@@ -0,0 +1,126 @@
+        with:
+          repository: 'cisco/openh264'
+          ref: openh264v2.3.1
+          path: openh264
+
+      - name: Build IMTL app dependency
+        run: | 
+         cd mman-win32
+          ./configure --prefix=${MSYSTEM_PREFIX}
+          make -j$(nproc) && make install
+          
+      - name: Build H264
+        run: |
+          cd openh264
+          meson build
+          ninja -C build install
+          
+        # Fixes in https://sourceforge.net/p/mingw-w64/mailman/message/29053757/
+      - name: Fix mingw64 binutils 
+        run: |
+          cp -f /mingw64/bin/ar.exe /mingw64/bin/x86_64-w64-mingw32-ar.exe
+          cp -f /mingw64/bin/dlltool.exe /mingw64/bin/x86_64-w64-mingw32-dlltool.exe
+          cp -f /mingw64/bin/nm.exe /mingw64/bin/x86_64-w64-mingw32-nm.exe
+          cp -f /mingw64/bin/strip.exe /mingw64/bin/x86_64-w64-mingw32-strip.exe
+          cp -f /mingw64/bin/windres.exe /mingw64/bin/x86_64-w64-mingw32-windres.exe
+          
+      - name: Checkout FFmpeg v4.4.2 aa28df
+        run: |
+          git config --global user.name github-actions
+          git config --global user.email github-actions@github.com
+          git clone https://git.ffmpeg.org/ffmpeg.git 
+          cd ffmpeg
+          git checkout release/4.4
+          git reset --hard aa28df74ab197c49a05fecc40c81e0f8ec4ad0c3
+          
+      - name: Apply IMTL patches
+        run: |
+          pushd ffmpeg
+          cp -f ../ecosystem/ffmpeg_plugin/kahawai_common.c ./libavdevice/
+          cp -f ../ecosystem/ffmpeg_plugin/kahawai_common.h ./libavdevice/
+          cp -f ../ecosystem/ffmpeg_plugin/kahawai_dec.c ./libavdevice/
+          cp -f ../ecosystem/ffmpeg_plugin/kahawai_enc.c ./libavdevice/
+          git am --whitespace=fix ../ecosystem/ffmpeg_plugin/0001-avdevice-kahawai-Add-kahawai-input-output-devices.patch
+    
+      - name: Setup FFmpeg enable MTL
+        run: >
+          cd ffmpeg &&
+          ./configure --arch=x86_64
+          --target-os=${{matrix.sys}}
+          --cross-prefix=x86_64-w64-mingw32- 
+          --prefix=${FFMPEG_PREFIX}
+          --enable-shared 
+          --enable-nonfree 
+          --enable-gpl 
+          --disable-lto 
+          --enable-pic  
+          --disable-w32threads 
+          --enable-mtl 
+          --enable-libx264 
+          --enable-libopenh264 
+          --enable-encoder=libopenh264
+       
+      - name: Build FFmpeg enable MTL
+        run: |
+          cd ffmpeg
+          make -j$(nproc)
+          make install
+          cat ffbuild/config.log
+          
+      - name: Checks FFmpeg enable MTL
+        run: |
+          cd ${FFMPEG_PREFIX}/bin
+          ./ffmpeg -version
+      
+      - name: Zip archive FFmpeg enable mtl
+        run: |
+          zip -r libmtl-${{matrix.sys}}-dpdk-v${{matrix.dpdk}}.zip build/lib
+          zip -r ffmpeg-mtl-${{matrix.sys}}-dpdk-v${{matrix.dpdk}}.zip $FFMPEG_PREFIX/bin
+      
+      - uses: actions/upload-artifact@v3
```

Comment:
Sorry I am new to `upload-artifact` action, so this pkg will be uploaded to where? And do we really need a upload for every PR?

Reply:
the upload-artifact basically will make the package available in the workflow runs once ffmpeg + IMTL built and checks successfully.

A sample of binary ffmpeg package: https://github.com/lab-github/Media-Transport-Library/actions/runs/5450822636

Reply:
The runtime artifacts will be uploaded on workflow job similar to the sample above
https://github.com/OpenVisualCloud/Media-Transport-Library/actions/runs/5451445708
and could be download for later consumption with ffmpeg.exe binary running MTL on windows machine